tests/ui/unsafe-fields/copy-trait.stderr

```diff
@@ -0,0 +1,28 @@
+error[E0200]: the trait `Copy` requires an `unsafe impl` declaration
+  --> $DIR/copy-trait.rs:13:5
+   |
+LL |     impl Copy for UnsafeEnum {}
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: the trait `Copy` cannot be safely implemented for `safe_impl::UnsafeEnum` because it has unsafe fields. Review the invariants of those fields before adding an `unsafe impl`
+help: add `unsafe` to this trait implementation
+   |
+LL |     unsafe impl Copy for UnsafeEnum {}
+   |     ++++++
+
+error[E0199]: implementing the trait `Copy` is not unsafe
```

Comment:
This means that this error is impossible to fix? Do you plan to improve that situation?

Reply:
How so? If the enum has unsafe fields, the fix is:
```diff
- impl Copy for UnsafeEnum {}
+ unsafe impl Copy for UnsafeEnum {}
```
If the enum has no unsafe fields, the fix is:
```diff
- unsafe impl Copy for SafeEnum {}
+ impl Copy for SafeEnum {}
```

Reply:
No, if you mark a copy impl as unsafe you will get E0199 since copy is not an unsafe trait. Please add a test for a working copy impl.

Reply:
[Done](https://github.com/rust-lang/rust/compare/d66fa52eb51b36863875294786b841919bc5b324..96fc9a9c59cdd41e6c003ff6409b176cc2264c8b). The ui test now contains both a working safe impl on a type without unsafe fields: https://github.com/rust-lang/rust/blob/96fc9a9c59cdd41e6c003ff6409b176cc2264c8b/tests/ui/unsafe-fields/copy-trait.rs#L7-L13

...and a working unsafe impl on a type with unsafe fields: https://github.com/rust-lang/rust/blob/96fc9a9c59cdd41e6c003ff6409b176cc2264c8b/tests/ui/unsafe-fields/copy-trait.rs#L25-L32